src/main/kotlin/JobState.kt

```diff
@@ -68,12 +73,19 @@ class JobState<U: Any> @PublishedApi internal constructor(val value: U?) {
         return status
     }
 
-    inline fun process(newStatus: JobStatus, predicate: (U) -> Boolean): JobState<U> {
-        if (inProgress() && !predicate(value!!))
+    inline fun process(newStatus: JobStatus, predicate: (U) -> Boolean, msg: String? = null): JobState<U> {
+        if (inProgress() && !predicate(value!!)) {
             this.status = newStatus
+            (msg ?: functionRef()).let { log.debug("Failed on: $it") }
```

Comment:
This leads to too many string and list allocations when not in debug mode.

```
log.debug {
 (msg ?: functionRef()).let { "Failed on $it" }
}
```

Reply:
You are right, that is a much better solution. However, while the whole functionRef was a nice idea - in theory - with suspend functions and inline functions (where all applicable functions are inline functions) you get references like this
>  Failed on: se.zensum.sms.status.WebKt$server$1.doResume(Web.kt:109)

where the line Web.kt:109 does not even exist. So the practical use is, as far as I can see, none :( I suspected this was the case, but I wanted it to try at least. If you know some other way to get similar functionality with inline functions, I would be more than happy to change the code. As it is now, I'll probably revert it to how it was in the previous commit.